### DISCUSSION TITLE
Is it possible to have MDsveX but having that data coming from a CMS

### DISCUSSION THREAD

**Discussion by itstheandre** [pngwn/MDsveX#395] [Q&A]
My question might be dumb, but I wonder if it is possible to, instead of having the data coming from internal files, to have the content coming from a CMS instead?

Is this possible? To introduce custom components to the mix as well? Or is this not a reality possible thing to do?

My use case:

I am going to be writing a website where I dont want to have to push the content to git everytime that I create an article, so I just want to put it in a headless CMS or having it stored in a personal db, for example.

My goal is to be able to do that and, if possible, to be able to use a collection of custom components with it as well.

In a blog article I would save something that would say which components I would want, but from that point I dont know if it is possible.

Thanks

**Comment by kharann** [comment]
Honestly, I would also prefer to retrieve markdown from a CMS. Currently, I'm using Sanity.io. I've managed to retrieve the data, but it feels somewhat hacky imo.

**Comment by Bandit** [comment]
Can you not use `compile`? https://mdsvex.com/docs#compile

**Comment by itstheandre** [comment]
To be honest the use case here would be to create something similar to what MDX does. Where you would be able to, theoretically could pass in to a config object maybe a list of components. 

I know you can already pass custom html tags, but the idea would be to eventually also pass custom components available in our application. This config object could be done on a load function or eventually at runtime.

**Comment by bwklein** [comment]
I've been enjoying this CMS option with my SvelteKit site. 
https://frontmatter.codes/

This way you are just editing markdown files in your project with a nice UI.

I run that within a GitPod.io workspace and it gives me a super portable editor and CMS in any browser.

**Reply by itstheandre** [reply]
but at the end of the day, this adds it to your filesystem right?

**Reply by zivbk1** [reply]
Yeah, just plain text files. No need for anything external to the project.

**Reply by itstheandre** [reply]
my original thinking would be somethign that wouldnt have to be necessarily connected with git. think of an external cms.
This way it still is the same import from MDsveX right? nothing too extraordinary, I would guess?

Do you still import components at the top of that file?

**Reply by bwklein** [reply]
Yes, although I am working on a way to push the component loading into the MDsveX layout for that content type and just add some structured data to the front matter that will be used to drive the page build and content, in addition to any components that you want to import as 'shortcodes' into the markdown itself.
